src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,295 @@
+    this.defineRestrictionsAndStick();
+    // If there's already a header being stick when the page is
+    // resized. The CSS style of the sticky-header may be not fit
+    // the resized window. So we need to unstuck it then re-stick it.
+    // unstuck() can set 'isStuck' to FALSE. Then stickElement() can work.
+    if (this.isStuck) {
+      this.unstuckElement();
+      this.stickElement();
+    }
+  }
+
+  /**
+   * define the restrictions of the sticky header(including stickyWidth,
+   * when to start, when to finish)
+   */
+  defineRestrictions(): void {
+    let containerTop: any = this.stickyParent.getBoundingClientRect();
+    let elemHeight: number = this.element.offsetHeight;
+    let containerHeight: number = this.getCssNumber(this.stickyParent, 'height');
+    this._containerStart = containerTop.top;
+
+    // the padding of the element being sticked
+    let elementPadding: any = this.getCssValue(this.element, 'padding');
+
+    let paddingNumber: any = Number(elementPadding.slice(0, -2));
+    this._scrollingWidth = this.upperScrollableContainer.clientWidth -
+      paddingNumber - paddingNumber;
+
+    this._scrollFinish = this._containerStart + (containerHeight - elemHeight);
+  }
+
+  /**
+   * Reset element to its original CSS
+   */
+  resetElement(): void {
+    this.element.classList.remove(STICK_START_CLASS);
+    Object.assign(this.element.style, this.originalCss);
+  }
+
+  /**
+   * Stuck element, make the element stick to the top of the scrollable container.
+   */
+  stickElement(): void {
+    this.isStuck = true;
+
+    this.element.classList.remove(STICK_END_CLASS);
+    this.element.classList.add(STICK_START_CLASS);
+
+    /**
+     * Have to add the translate3d function for the sticky element's css style.
+     * Because iPhone and iPad's browser is using its owning rendering engine. And
+     * even if you are using Chrome on an iPhone, you are just using Safari with
+     * a Chrome skin around it.
+     *
+     * Safari on iPad and Safari on iPhone do not have resizable windows.
+     * In Safari on iPhone and iPad, the window size is set to the size of
+     * the screen (minus Safari user interface controls), and cannot be changed
+     * by the user. To move around a webpage, the user changes the zoom level and position
+     * of the viewport as they double tap or pinch to zoom in or out, or by touching
+     * and dragging to pan the page. As a user changes the zoom level and position of the
+     * viewport they are doing so within a viewable content area of fixed size
+     * (that is, the window). This means that webpage elements that have their position
+     * "fixed" to the viewport can end up outside the viewable content area, offscreen.
+     *
+     * So the 'position: fixed' does not work on iPhone and iPad. To make it work,
+     * 'translate3d(0,0,0)' needs to be used to force Safari re-rendering the sticky element.
+     **/
+    this.element.style.transform = 'translate3d(0,0,0)';
```

Comment:
I believe IE11 will have trouble with this, change to `translate3d(0px, 0px, 0px);